Wikipedia/Code/WMFSuggestedEditsContentSource.m

```diff
@@ -24,42 +24,45 @@ - (instancetype)initWithDataStore:(MWKDataStore *)dataStore {
 
 - (void)loadNewContentInManagedObjectContext:(nonnull NSManagedObjectContext *)moc force:(BOOL)force completion:(nullable dispatch_block_t)completion {
 
-    // First delete old card
-    [self removeAllContentInManagedObjectContext:moc];
-
-    NSURL *appLanguageSiteURL = self.dataStore.languageLinkController.appLanguage.siteURL;
-    WMFAuthenticationManager *authManager = self.dataStore.authenticationManager;
-
-    if (!appLanguageSiteURL) {
-        completion();
-        return;
-    }
-    
-    if (!authManager.authStateIsPermanent) {
-        completion();
-        return;
-    }
+    [moc performBlock:^{
+        // First delete old card
+        [self removeAllContentInManagedObjectContext:moc];
         
-    WMFCurrentUser *user = [self.dataStore.authenticationManager userWithSiteURL:appLanguageSiteURL];
-    
-    // Image Recommendations Business Logic:
-    // Do not show suggested edits option if users have < 50 edits or they have VoiceOver on.
-
-    BOOL isEligibleForImageRecommendations = (user && user.editCount > 50 && !user.isBlocked && !UIAccessibilityIsVoiceOverRunning());
-
-    if ([self isEligibleForAltText:user] || isEligibleForImageRecommendations) {
-        [self.growthTasksDataController hasImageRecommendationsWithCompletion:^(BOOL hasRecommendations) {
-            if (hasRecommendations) {
-                NSURL *URL = [WMFContentGroup suggestedEditsURLForSiteURL:appLanguageSiteURL];
-
-                [moc fetchOrCreateGroupForURL:URL ofKind:WMFContentGroupKindSuggestedEdits forDate:[NSDate date] withSiteURL:appLanguageSiteURL associatedContent:nil customizationBlock:nil];
-            }
+        NSURL *appLanguageSiteURL = self.dataStore.languageLinkController.appLanguage.siteURL;
+        WMFAuthenticationManager *authManager = self.dataStore.authenticationManager;
 
+        if (!appLanguageSiteURL) {
             completion();
-        }];
-    } else {
-        completion();
-    }
+            return;
+        }
+        
+        if (!authManager.authStateIsPermanent) {
+            completion();
+            return;
+        }
+            
+        WMFCurrentUser *user = [self.dataStore.authenticationManager userWithSiteURL:appLanguageSiteURL];
```

Comment:
On a fresh install, the user returns nil when I change language at this point. If I log out and then log in again, the user is successfully retrieved, and then the card appears. 
What I did:
- Fresh install
- App auto login happens.
- English as primary language - user is not nil
- Add PT as primary language (I've also tested with other languages that are on the alt text experiment, such as FR and ES)
- User returns nil from authenticationManager userWithSiteURL

Let me know if you reproduce that.